app/services/service_listeners/attachment_updater.rb

```diff
@@ -5,56 +5,57 @@ def initialize(attachable)
     end
 
     def update!
-      Attachment.where(attachable: attachable.attachables).find_each do |attachment|
+      Attachment.where(attachable: @attachable.attachables).find_each do |attachment|
         next unless attachment.attachment_data
-
-        draft_status_updater attachment.attachment_data
-        redirect_url_updater attachment.attachment_data
-        link_header_updater attachment.attachment_data
-        access_limited_updater attachment.attachment_data
-        deleter attachment.attachment_data
-
-        AttachmentData.where(replaced_by_id: attachment.attachment_data.id).find_each do |attachment_data|
-          replacement_id_updater attachment_data
-        end
+        self.class.update_attachment_data! attachment.attachment_data
       end
     end
 
-  private
+    def self.update_attachment_data!(attachment_data)
```

Comment:
It might be nicer instead to have the interface for this class to work something like this:

```ruby
ServiceListeners::AttachmentUpdater.call(attachable: attachable)
ServiceListeners::AttachmentUpdater.call(attachment_data: attachment_data)
```